test/cli/transports_test.go

```diff
@@ -125,3 +125,36 @@ func TestTransports(t *testing.T) {
 	})
 
 }
+
+func TestStartsWebTransport(t *testing.T) {
+	t.Run("announces without listening on web transport", func(t *testing.T) {
+		t.Parallel()
+		n := harness.NewT(t).NewNode().Init()
+		n.UpdateConfig(func(cfg *config.Config) {
+			cfg.Addresses.Swarm = []string{"/ip4/127.0.0.1/udp/0/quic-v1"}
+			cfg.Addresses.Announce = []string{"/ip4/127.0.0.1/udp/0/quic-v1/webtransport"}
+		})
+		n.StartDaemon()
+
+		// If we announce WebTransport, but do not listen on it, there will NOT
+		// be a certhash since the certificate manager is not initialized.
+		output := string(n.Daemon.Stdout.Bytes())
+		assert.Contains(t, output, "/webtransport")
+		assert.NotContains(t, output, "certhash")
```

Comment:
> I'm not really sure the intuition here, yes this represents the result of what the PR will currently do. owever, this will not doing anything useful for the user (i.e. there is no value gained from advertising an announce address that we can't possibly be listening on).

Well, this tests would've failed in `master`. And no, there is no usefulness to announce something that we're not listening to. We can change that. But I'd argue that's outside of the scope of this PR. That's Kubo's current behaviour. The idea here is to fix it and unblock the release. If we want to stop allowing announcing things we're not listening to, we can open a new issue and address that separetly.

> In any event if we keep this test here we should either just test that another node can dial on quic-v1 after the daemon has started (instead of checking the returned addresses) or add some comments here about it being ok to delete/modify this part of the test since it's effectively testing go-libp2p/kubo internal implementation details that don't impact users

I will address that in a bit.